src/components/CurrencyInput.tsx

```diff
@@ -135,7 +135,12 @@ export const CurrencyInput: FC<CurrencyInputProps> = forwardRef<
         return;
       }
 
-      const numberValue = parseFloat(stringValue.replace(decimalSeparator, '.'));
+      const numberValue = (() => {
+        const stringValueWithoutSeparator = decimalSeparator
+          ? stringValue.replace(decimalSeparator, '.')
+          : stringValue;
+        return parseFloat(stringValueWithoutSeparator);
+      })();
```

Comment:
I'm personally not a big fan of immediately invoking a function for something like this.

Perhaps:

```js
const stringValueWithoutSeparator = decimalSeparator
        ? stringValue.replace(decimalSeparator, '.')
        : stringValue;
const numberValue = parseFloat(stringValueWithoutSeparator);
```

Reply:
I have removed the IIFE.